core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -320,7 +336,8 @@ private[spark] class SparkSubmit extends Logging {
       }
     }
 
-    args.sparkProperties.foreach { case (k, v) => sparkConf.set(k, v) }
+    // update spark config from args
+    args.toSparkConf(Option(sparkConf))
```

Comment:
Need to keep this here as `args` are being updated within the body of this method. We should clean it up in another PR. args should be immutable and we should only modify SparkConf.